esp/services/ws_ecl/ws_ecl_service.hpp

```diff
@@ -87,11 +87,21 @@ typedef enum wsEclTypes_
 
 } wsEclType;
 
+class RoxieConnEndpoint : public CSmartSocketFactory
+{
+public:
+    bool includeTargetInURL;
+
+    RoxieConnEndpoint(const char *_socklist, bool _retry, bool includeTarget) : CSmartSocketFactory(_socklist, _retry), includeTargetInURL(includeTarget)
+    {
+    }
+};
+
 class CWsEclService : public CInterface,
     implements IEspService
 {
 public:
-    MapStringToMyClass<ISmartSocketFactory> connMap;
+    MapStringToMyClassViaBase<RoxieConnEndpoint, ISmartSocketFactory> connMap;
```

Comment:
CSmartSocketFactory uses multiple inheritance.  Previously mapping to interface ISmartSocketFactory cast away the multiple inheritance.  New class, not so much.  But I'll change to using a single static_cast where I need it instead.